flocker/node/agents/test/test_blockdevice.py

```diff
@@ -3395,23 +3478,99 @@ class CreateBlockDeviceDatasetInterfaceTests(
     """
 
 
-class CreateBlockDeviceDatasetImplementationTests(SynchronousTestCase):
+class CreateBlockDeviceDatasetImplementationMixin(object):
     """
-    ``CreateBlockDeviceDataset`` implementation tests.
+    Utility Mixin for ``CreateBlockDeviceDataset`` implementation tests.
     """
-    def setUp(self):
-        self.api = loopbackblockdeviceapi_for_test(
-            self,
-            allocation_unit=LOOPBACK_ALLOCATION_UNIT
+
+    def _create_blockdevice_dataset(self, dataset_id, maximum_size,
+                                    metadata=pmap({})):
+        """
+        Call ``CreateBlockDeviceDataset.run`` with a ``BlockDeviceDeployer``.
+
+        :param UUID dataset_id: The uuid4 identifier for the dataset which will
+            be created.
+        :param int maximum_size: The size, in bytes, of the dataset which will
+            be created.
+        :param pmap(unicode, unicode) metadata: The metadata for the dataset.
+
+        :returns: A 3-tuple of:
```

Comment:
:grinning: yea this was just moving code. Although I agree that PClass is better, and I like small incremental improvements to code so I'll go back and change all callsites before merging.